store.go

```diff
@@ -19,7 +21,39 @@ func NewEngineJIT() *Engine { // TODO: compiler?
 	return &Engine{e: jit.NewEngine()}
 }
 
-// HostFunctions are functions written in Go, which a WebAssembly Module can import.
+// StoreConfig allows customization of a Store via NewStoreWithConfig
+type StoreConfig struct {
+	// Engine defaults to NewEngineInterpreter
+	Engine *Engine
```

Comment:
do we need to export?

Reply:
oh this is config! nvm

Reply:
nope!

Reply:
well otherwise an end-user can't choose an engine, so yeah I think we need to keep this exported.